src/topology/local_homeomorph.lean

```diff
@@ -239,38 +252,165 @@ lemma map_nhds_within_preimage_eq (e : local_homeomorph α β) {x} (hx : x ∈ e
 by rw [e.map_nhds_within_eq hx, e.image_source_inter_eq', e.target_inter_inv_preimage_preimage,
   e.nhds_within_target_inter (e.map_source hx)]
 
+lemma preimage_open_of_open {s : set β} (hs : is_open s) : is_open (e.source ∩ e ⁻¹' s) :=
+e.continuous_on.preimage_open_of_open e.open_source hs
+
+/-- We say that `t : set β` is an image of `s : set α` under a local homeomorphism `e` if any of the
+following equivalent conditions hold:
+
+* `e '' (e.source ∩ s) = e.target ∩ t`;
+* `e.source ∩ e ⁻¹ t = e.source ∩ s`;
+* `∀ x ∈ e.source, e x ∈ t ↔ x ∈ s` (this one is used in the definition).
+-/
```

Comment:
All this makes sense for local equivs, as well as a lot of the following lemmas. Would it make sense to register first the API for local equivs?

Reply:
https://leanprover-community.github.io/mathlib_docs/data/equiv/local_equiv.html#local_equiv.is_image

Reply:
I added a section header and mentioned `local_equiv.is_image` there.

Reply:
🤦 😄